azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServerModule.java

```diff
@@ -87,8 +89,15 @@ protected void configure() {
   }
 
   private Class<? extends ExecutorManagerAdapter> resolveExecutorManagerAdaptorClassType() {
-    return this.props.getBoolean(ConfigurationKeys.AZKABAN_POLL_MODEL, false)
-        ? ExecutionController.class : ExecutorManager.class;
+    switch(DispatchMethod.getDispatchModel(this.props)) {
```

Comment:
What do you think of reading the `azkaban.execution.dispatch.method` at initialization time here to inject the right class and defining a `getDispatchMethod` method in the ExecutorManagerAdapter interface which will then be implemented as `return DispatchMethod.XXX` by every class extending ExecutorManagerAdapter.

Reply:
I have made that change for web server. For executor server, property and getDispatchMethod is needed. I have changed the method definition. It won't take props object anymore.

Reply:
Sorry, I didn't realize that most of the usages of `getDispatchMethod` were on the executor module were there is no ExecutorManagerAdapter class available. Then, for now, there is no need for a "getDispatchMethod" in the that interface because once the right class is injected branching logic based on dispatch model isn't needed.

Reply:
Yes agree. I am using getDispatchMethod for now in Test case to validate the correct class is injected based on property. So let's keep that method.